tests/test_unit/test_transforms.py

```diff
@@ -0,0 +1,205 @@
+    [
+        pytest.param(
+            {},
+            data_array_with_dims_and_coords(nparray_0_to_23()),
+            id="Do nothing",
+        ),
+        pytest.param(
+            {"unit": "elephants"},
+            data_array_with_dims_and_coords(
+                nparray_0_to_23(), unit="elephants"
+            ),
+            id="No scaling, add unit",
+        ),
+        pytest.param(
+            {"factor": 2},
+            data_array_with_dims_and_coords(nparray_0_to_23() * 2),
+            id="Double, no unit",
+        ),
+        pytest.param(
+            {"factor": 0.5},
+            data_array_with_dims_and_coords(nparray_0_to_23() * 0.5),
+            id="Halve, no unit",
+        ),
+        pytest.param(
+            {"factor": 0.5, "unit": "elephants"},
+            data_array_with_dims_and_coords(
+                nparray_0_to_23() * 0.5, unit="elephants"
+            ),
+            id="Halve, add unit",
+        ),
+        pytest.param(
+            {"factor": [0.5, 2]},
+            data_array_with_dims_and_coords(
+                nparray_0_to_23() * [0.5, 2],
+            ),
+            id="x / 2, y * 2",
+        ),
+        pytest.param(
+            {"factor": np.array([0.5, 2]).reshape(1, 2)},
+            data_array_with_dims_and_coords(
+                nparray_0_to_23() * [0.5, 2],
+            ),
+            id="x / 2, y * 2, should squeeze to cast across space",
+        ),
+    ],
+)
+def test_scale(
+    sample_data: xr.DataArray,
+    optional_arguments: dict[str, Any],
+    expected_output: xr.DataArray,
+):
+    """Test scaling with different factors and units."""
+    scaled_data = scale(sample_data, **optional_arguments)
+    xr.testing.assert_equal(scaled_data, expected_output)
+    assert scaled_data.attrs == expected_output.attrs
+
+
+def test_scale_inverted_data():
+    """Test scaling with transposed data along the correct dimension.
+
+    The factor is reshaped to (1, 1, 4, 1) so that it can be broadcasted along
+    the third dimension ("y") which matches the length of the scaling factor.
+    """
+    factor = [0.5, 2]
+    transposed_data = data_array_with_dims_and_coords(
+        nparray_0_to_23().transpose(), dims=["space", "time"]
+    )
+    output_array = scale(transposed_data, factor=factor)
+    expected_output = data_array_with_dims_and_coords(
+        (nparray_0_to_23() * factor).transpose(), dims=["space", "time"]
+    )
+    xr.testing.assert_equal(output_array, expected_output)
+
+    factor = [0.1, 0.2, 0.3, 0.4]
+    data_shape = (3, 5, 4, 2)
+    numerical_data = np.arange(np.prod(data_shape)).reshape(data_shape)
+    input_data = xr.DataArray(numerical_data, dims=["w", "x", "y", "z"])
+    output_array = scale(input_data, factor=factor)
+    assert output_array.shape == input_data.shape
+    xr.testing.assert_equal(
+        output_array, input_data * np.array(factor).reshape(1, 1, 4, 1)
+    )
+
+
+def test_scale_first_matching_axis():
```

Comment:
This test also needs to be modifies or binned, if you adopt my suggestion about specifically vroadcasting over `space`.